sdks/java/io/splunk/src/main/java/org/apache/beam/sdk/io/splunk/SplunkEventWriter.java

```diff
@@ -396,6 +411,41 @@ private static void flushWriteFailures(
     }
   }
 
+  /**
+   * Reads a root CA certificate from GCS and returns it as raw bytes.
+   *
+   * @param filePath path to root CA cert in GCS
+   * @return raw contents of cert
+   * @throws RuntimeException thrown if not able to read or parse cert
+   */
+  public static byte[] getCertFromGcsAsBytes(String filePath) throws IOException {
+    ReadableByteChannel channel = getGcsFileByteChannel(filePath);
+    try (InputStream inputStream = Channels.newInputStream(channel)) {
+      return IOUtils.toByteArray(inputStream);
+    } catch (IOException e) {
+      throw new RuntimeException("Error when reading: " + filePath, e);
+    }
+  }
+
+  /** Handles getting the {@link ReadableByteChannel} for {@code filePath}. */
+  private static ReadableByteChannel getGcsFileByteChannel(String filePath) throws IOException {
```

Comment:
Is filePath expected to be a glob (not the exact path) ?

If it's an exact file path, this matching and verification is unnecessary and we can just assume it to be a single file and do "FileSystems.open" (and remove this method).

Reply:
Yeah, the cert is expected to be a single path. Decided to use `FileSystems.matchSingleFileSpec(filePath)` instead, since this also checks if the file path is specifically a single file